core/dronecore.cpp

```diff
@@ -23,6 +23,82 @@ DroneCore::~DroneCore()
     _impl = nullptr;
 }
 
+DroneCore::ConnectionResult DroneCore::add_any_connection(const std::string &connection_url)
+{
+    std::string delimiter = "://";
+    std::string conn_url(connection_url);
+    std::vector<std::string> connection_str;
+    size_t pos = 0;
+    /*Parse the connection url to get Protocol, IP or Serial dev node
+      and port number or baudrate as per the URL definition
+    */
+    for (int i = 0; i < 2; i++) {
+        pos = conn_url.find(delimiter);
+        if (pos != std::string::npos) {
+            connection_str.push_back(conn_url.substr(0, pos));
+            // Erase the string which is parsed already
+            conn_url.erase(0, pos + delimiter.length());
+            if (conn_url == "") {
+                break;
+            }
+            delimiter = ":";
+        }
+    }
+    connection_str.push_back(conn_url);
+    /* check if the protocol is Network protocol or Serial */
+    if (connection_str.at(0) != "serial") {
+        int port = 0;
+        if (connection_str.at(2) != "") {
+            port = std::stoi(connection_str.at(2));
+        }
+        return add_link_connection(connection_str.at(0), connection_str.at(1),
+                                   port); //Network Protocol : TCP or UDP
+    } else if (connection_str.at(0) == "serial") {
+        if (connection_str.at(1) == "") {
+            return add_serial_connection();
+        } else {
+            return add_serial_connection(connection_str.at(1), std::stoi(connection_str.at(2)));
+        }
+    }
+
+    return DroneCore::ConnectionResult::CONNECTION_URL_INVALID;
+}
+
+/* Creates a connection link only for the Network protocols : UDP and TCP*/
+DroneCore::ConnectionResult DroneCore::add_link_connection(const std::string &protocol,
```

Comment:
This method needs to be in `dronecore_impl.cpp` in the same way as others like `add_udp_connection()`.

Reply:
add_link_connection is calling add_udp_connection() and add_tcp_connection() which are not accessible in dronecore_impl. If i modify it to add in dronecore_impl, it'll duplicate add_udp_connection and add_tcp_connection.

Reply:
Oh you are right. I think it's actually wrong that `add_udp_connection()` etc. are implemented in `dronecore.cpp` as well. They should be implemented in `dronecore_impl.cpp`. In that case if you want to move everything that would be great but otherwise I'm also ok with merging it as is, and clean it up later.